src/main/java/com/mraof/minestuck/item/weapon/MusicPlayerItem.java

```diff
@@ -0,0 +1,146 @@
+package com.mraof.minestuck.item.weapon;
+
+import com.mraof.minestuck.block.EnumCassetteType;
+import com.mraof.minestuck.client.sounds.MusicPlayerOnPlayerSoundInstance;
+import com.mraof.minestuck.inventory.musicplayer.CapabilityMusicPlaying;
+import com.mraof.minestuck.inventory.musicplayer.CapabilityProviderMusicPlayer;
+import com.mraof.minestuck.inventory.musicplayer.IMusicPlaying;
+import com.mraof.minestuck.inventory.musicplayer.MusicPlayerContainer;
+import com.mraof.minestuck.item.CassetteItem;
+import com.mraof.minestuck.util.Debug;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.sounds.SoundManager;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResultHolder;
+import net.minecraft.world.SimpleMenuProvider;
+import net.minecraft.world.effect.MobEffect;
+import net.minecraft.world.effect.MobEffectInstance;
+import net.minecraft.world.effect.MobEffects;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.ai.attributes.AttributeInstance;
+import net.minecraft.world.entity.ai.attributes.Attributes;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.items.CapabilityItemHandler;
+import net.minecraftforge.items.IItemHandler;
+import net.minecraftforge.network.NetworkHooks;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Random;
+
+public class MusicPlayerItem extends WeaponItem
+{
+	private MobEffect effect = null;
+	private float applyingChance = 0.0F;
+	int duration = 0;
```

Comment:
You should not be using mutable fields on items. Use some different means for passing these around.